src/views/ToDoList/ToDoList.test.tsx

```diff
@@ -0,0 +1,262 @@
+          selector: 'p',
+        });
+        const errorParagraph = toDoList.queryByTestId('comment-error');
+
+        expect(firstTasksParagraph).toBeDefined();
+        expect(secondTasksParagraph).toBeDefined();
+        expect(thirdTasksParagraph).toBeDefined();
+        expect(fourthTasksParagraph).toBeDefined();
+        expect(errorParagraph).toBeNull();
+      });
+    });
+    describe('and the Api call is unsuccessful', () => {
+      beforeEach(() => {
+        (toDosApi.fetchToDos as Mock).mockRejectedValue(null);
+      });
+      it('should display error and not display tasks', async () => {
+        const toDoList = render(<ToDoList />);
+
+        const errorParagraph = await toDoList.findByText(
+          'Failed to catch tasks list',
+          { selector: 'p' },
+        );
+        const firstTaskParagraph = toDoList.queryByText(task1.title, {
+          selector: 'p',
+        });
+
+        expect(errorParagraph).toBeDefined();
+        expect(firstTaskParagraph).toBeNull();
+      });
+    });
+  });
+  describe('when user types the title', () => {
+    describe('and clicks the submit button', () => {
+      beforeEach(() => {
+        (toDosApi.fetchToDos as Mock).mockResolvedValue(returnedTasksList);
+        (toDosApi.createNewTask as Mock).mockReturnValue(
+          new Promise(() => null),
+        );
+      });
+      it('should make attempt to create the POST request', () => {
+        const toDoList = render(<ToDoList />);
+
+        const titleInput = toDoList.getByPlaceholderText('title');
+
+        fireEvent.input(titleInput, { target: { value: newTaskTitle } });
+
+        const button = toDoList.getByRole('button');
+
+        fireEvent.click(button);
+
+        expect(toDosApi.createNewTask).toHaveBeenCalledWith(newTaskMock);
+      });
+      describe('if the Api call is successful', () => {
+        beforeEach(() => {
+          (toDosApi.createNewTask as Mock).mockResolvedValue(newTaskMock);
+        });
+        it('should create new task and not throw error', async () => {
+          const toDoList = render(<ToDoList />);
+
+          const titleInput = toDoList.getByPlaceholderText('title');
+
+          fireEvent.change(titleInput, { target: { value: newTaskTitle } });
+
+          const button = toDoList.getByRole('button');
+
+          fireEvent.click(button);
+
+          const newTaskParagraph = await toDoList.findByText(newTaskTitle, {
+            selector: 'p',
+          });
+
+          const errorParagraph = toDoList.queryByText(
+            'Failed to catch tasks list',
+            { selector: 'p' },
+          );
+
+          expect(newTaskParagraph).toBeDefined();
+          expect(errorParagraph).toBeNull();
+        });
+      });
+      describe('if the Api call is unsuccessful', () => {
+        beforeEach(() => {
+          (toDosApi.createNewTask as Mock).mockRejectedValue(null);
+        });
+        it('should throw an error and not create task', async () => {
```

Comment:
`throw error` is a bit vague. Here you are checking if the error is **rendered**

Also,
- not displaying the created task
- displaying an error
are two separate test cases and should have two separate its